main/main_views/order_detail.py

```diff
@@ -0,0 +1,12 @@
+from rest_framework import generics
+from main.models import Order
+from main.serializers import OrderDetailSerializer
+
+
+# TODO: Change to Retrieve API View
+class OrderDetail(generics.RetrieveUpdateDestroyAPIView):
```

Comment:
Please change to RetrieveAPIView as we don't want the ability to change and order from a non-admin type view. Also, please add authentication class and permission class to the view.